File: Python/Find the Index of the First Occurrence in a String.py
class Solution(object):
    def strStr(self, haystack, needle):
        """
        :type haystack: str
        :type needle: str
        :rtype: int
        """
        n=len(needle)
        window=0
        # window=haystack[:n]
        for i in range(0,len(haystack)-n+1):
            window= haystack[i:i+n]
            if window==needle:
                return i

        return -1

File: Java/Find the Index of the First Occurrence in a String.java
class Solution {
    public int strStr(String haystack, String needle) {
        int n=needle.length()-1;
        for(int i=n;i<haystack.length();i++){
            if(needle.equals(haystack.substring(i-n,i+1))){
                return i-n;
            }
        }
        return -1;
        
    }
}
